flatsurf/geometry/surface.py

```diff
@@ -18,64 +70,75 @@
+from six.moves import range
 from six import iteritems
+from collections import deque
 
 from sage.structure.sage_object import SageObject
-from sage.sets.family import Family
+
 
 class Surface(SageObject):
     r"""
-    An oriented surface built from a set of polygons and edges identified with
-    similarities (i.e. composition of homothety, rotations and translations).
+    Abstract base class of all surfaces that are built from a set of polygons
+    with edges identified. The identifications are compositions of homothety,
+    rotations and translations, i.e., similarities that ensure that the surface
+    is oriented.
+
+    Concrete implementations of a surface must implement at least
+    :meth:`polygon` and :meth:`opposite_edge`.
+
+    To be able to modify a surface, subclasses should also implement
+    :meth:`_change_polygon`, :meth:`_set_edge_pairing`, :meth:`_add_polygon`,
+    :meth:`_remove_polygon`.
+
+    For concrete implementations of a Surface, see, e.g., :class:`Surface_list`
+    and :class:`Surface_dict`.
+
+    INPUT:
+
+    - ``base_ring`` -- the ring containing the coordinates of the vertices of
+      the polygons
+
+    - ``base_label`` -- the label of a chosen special polygon in the surface,
+      see :meth:`base_label`
 
-    Each polygon is identified with a unique key (its label). The choice of the
-    label of the polygons is done at startup. If the set is finite then by
-    default the labels are the first non-negative integers 0,1,...
+    - ``finite`` -- whether this is a finite surface, see :meth:`is_finite`
 
-    The edge are identified by a couple (polygon label, edge number).
+    - ``mutable`` -- whether this is a mutable surface; can be changed later
+      with :meth:`set_immutable`
 
-    .. NOTE::
+    EXAMPLES::
+
+        sage: from flatsurf.geometry.surface import Surface, Surface_list, Surface_dict
 
-        This class is abstract and should not be called directly. Instead you
-        can either use Surface_list or Surface_dict. Another option is to
-        inherit from Surface and implement the two methods:
+        sage: S = Surface_list(QQ)
+        sage: isinstance(S, Surface)
+        True
 
-        - polygon(self, label): the polygon associated to the provided label
-        - opposite_edge(self, label, edge): a couple (``other_label``, ``other_edge``)
-          representing the edge being glued
+        sage: S = Surface_dict(QQ)
+        sage: isinstance(S, Surface)
+        True
 
-        If you want the surface to be mutable, you should override the methods:
-        - _change_polygon(self, label, new_polygon, gluing_list=None)
-        - _set_edge_pairing(self, label1, edge1, label2, edge2)
-        - _add_polygon(self, new_polygon, gluing_list=None, label=None)
-        - _remove_polygon(self, label)
-        See the documentation of those methods for details.
     """
-    def __init__(self, base_ring, base_label, finite=None, mutable=False):
```

Comment:
I changed this so all parameters are required now. `finite=None` was not a valid default actually since it led to a `ValueError` in line 74.